packages/dds/tree/docs/main/merge-semantics.md

```diff
@@ -0,0 +1,269 @@
+
+### Minimal Preconditions
+
+In the context of merge semantics,
+preconditions can be used to prevent an edit/transaction from being applied
+if concurrent edits that were sequenced earlier have impacted the document in a way that violates some specific requirements.
+
+By default, that is, in the absence of [constraints](#constraints),
+the only precondition that must be met in order for an edit to apply,
+is that the edit must not violate the document's schema.
+
+For example,
+when moving a node to a specific destination only requires that the node's type be allowed by the schema of the destination.
+
+One could imagine a host of potential additional preconditions for such an edit:
+
+-   The node being moved must not have been concurrently moved
+-   The node being moved must not have been concurrently removed
+-   Ancestors of the node being moved must not have been concurrently removed
+-   The parent and ancestors of destination must not have been concurrently removed
+
+`SharedTree`'s merge semantics are such that, by default, only the schema matters.
+The benefits of not violating the schema are obvious,
+but the benefits of not having more preconditions are more subtle.
+Indeed, while no one should wish for their edit to violate the schema,
+one may wish for their edit not to be carried out if some other precondition is not met.
+`SharedTree` does make that possible, but this is achieved bu using opt-in [constraints](#constraints).
+This effectively makes each editing API parameterizable so that any combination of preconditions is achievable.
+On the other hand, if `SharedTree`'s edits had some additional preconditions by default,
+then there would be no way for users to opt-out of these preconditions,
+thereby limiting their ability to express a more permissive edit.
+
+In the rest of this section, we call out some interesting implications that follow from the doctrine of minimal preconditions,
+the last of which is also a key benefit of this doctrine.
+
+#### Removal Is Movement
+
+`SharedTree` allows subtrees to be removed,
+such as when an element from an array node is removed,
+a key is deleted from a map node,
+or when the field on an object is overwritten.
+This removal does not adversely affect concurrent edits that target the removed subtree.
+
+Consider the following scenario:
+Alice removes a whole list of sticky notes, while Bob concurrently moves a sticky note out of that list and into another (non-removed) list.
+`SharedTree`'s removal semantics ensure that Bob's edit will apply no matter the sequencing order.
+If that weren't the case, then there would be a race between Alice and Bob's edit,
+where Bob's edit would not apply if Alice's edit were sequenced first.
+
+Note that in a lot of cases, the changes to the removed subtree won't be immediately visible because the tree is removed.
+They will, however, become visible if that removal is undone.
```

Comment:
You're not wrong, but the thing is:
1. We don't yet have that doc
2. I do mean undone in the most general sense (for example, the element or one of its ancestors could be moved back into the tree)
3. This is a case where it seems okay to rely on the reader's intuitive understanding of the term because the specifics are not material to the point being made.